pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -251,7 +272,169 @@ def measure_final_state(self, circuit: QuantumScript) -> Result:
             Tuple[TensorLike]: The measurement results
         """
 
+        if not circuit.shots:
+            # analytic case
+            if len(circuit.measurements) == 1:
+                return self.measurement(circuit.measurements[0])
+
+            return tuple(self.measurement(mp) for mp in circuit.measurements)
+
+        # finite-shot case
+        results = self.measure_with_samples(
+            circuit.measurements,
+            shots=circuit.shots,
+        )
+
         if len(circuit.measurements) == 1:
-            return self.measurement(circuit.measurements[0])
+            if circuit.shots.has_partitioned_shots:
+                return tuple(res[0] for res in results)
```

Comment:
when is it triggered?

Reply:
When one has something like `dev = qml.device(name, shots=[10, 20, 30])`.